user-operator/src/main/java/io/strimzi/operator/user/operator/SimpleAclOperator.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator;
+
+import io.strimzi.operator.common.operator.resource.ReconcileResult;
+import io.strimzi.operator.user.model.KafkaUserModel;
+import io.strimzi.operator.user.model.acl.SimpleAclRule;
+import io.strimzi.operator.user.model.acl.SimpleAclRuleResource;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import kafka.security.auth.Acl;
+import kafka.security.auth.Resource;
+import kafka.security.auth.SimpleAclAuthorizer;
+import org.apache.kafka.common.security.auth.KafkaPrincipal;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import scala.Tuple2;
+import scala.collection.Iterator;
+
+/**
+ * SimlpeAclOperator is responsible for managing the authorization rules in Apache Kafka / Apache Zookeeper.
+ * It is using Kafka's SimpleAclAuthorizer class to interact with Zookeeper and manage Acl rules.
+ * Since SimpleAclAuthorizer is written in Scala, this operator is using some Scala structures required for passing to / returned from the SimpleAclAuthorizer object.
+ * This class expects the SimpleAclAuthorizer instance to be passed from the outside.
+ * That is useful for testing and is similar to how the Kubernetes client is passed around.
+ */
+public class SimpleAclOperator {
```

Comment:
This is not a resource operator. Not all operators extend `AbstractResourceOperator`. We have the workaround operators for RBAC. And we have the _assembly_ operators which are IMHO much closer to this.

I do not think there is really a common interface for all operators. For example all have the `reconcile` method. But the signatures are very different. The `SimpleAclOperator` for example takes lists as parameters. So the only common class with the resource operators would be probably `Object`.

Additionally, we do not really use the abstraction for the purpose of using the operators - we always use the resource operators directly. We use it only to avoid code duplication. So I do not think having an interface which would have `Object` everywhere and not much in common would really help. We would need to redesign the way we use them to actually get some value from it.